### PR TITLE
key changes to blockchains files and tests

### PR DESCRIPTION
**Pull request by nzoghb**
Should be pretty straightforward, accounts for recent key changes in `dagora-chain`

### CODE REVIEW

**Comment by nzoghb** [comment]
most of the blockchain-related errors require the `dagora-chain` branch to be merged first

**Comment by kiddyboots216** [comment]
Lot of whitespace changes, what are those a result of?

File: core/fed_avg_optimizer.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def __init__(self, initialization_payload, dataset_manager):
 		max_rounds(int): how many rounds of fed learning this optimizer wants to do
 		"""
 		logging.info("Setting up Optimizer...")
+		# initialization_payload = initialization_payload.get_tx()
```
Comment (neeleshdodda44) [inline core/fed_avg_optimizer.py]: Did you mean to leave this commented out?

File: core/utils/dmljob.py [hunk 2]
```diff
@@ -24,7 +24,8 @@ def __init__(
         uuid=None,
         raw_filepath=None,
         session_filepath=None,
-        datapoint_count=None
+        datapoint_count=None,
+        round_num=None,
```
Comment (neeleshdodda44) [inline core/utils/dmljob.py]: Could you explain what `round_num` is?

File: core/utils/dmljob.py [hunk 3]
```diff
@@ -129,7 +132,7 @@ def deserialize_job(serialized_job):
         rest['uuid'],
         rest['raw_filepath'],
         rest['session_filepath'],
-        rest['datapoint_count']
+        rest['datapoint_count'],
```
Comment (neeleshdodda44) [inline core/utils/dmljob.py]: Did you mean to add `round_num` here too?

File: tests/testing_utils.py [hunk 4]
```diff
@@ -11,7 +11,8 @@ def make_communicate_job(key, weights):
     communicate_job = DMLJob(
         JobTypes.JOB_COMM.name,
         key,
-        "keras"
+        "keras",
+        round_num=420,
```
Comment (neeleshdodda44) [inline tests/testing_utils.py]: ...

File: core/utils/dmljob.py (hunk 2 above)
Comment (neeleshdodda44) [reply core/utils/dmljob.py]: Or actually can you put it in the docstring?

File: core/fed_avg_optimizer.py (hunk 1 above)
Comment (kiddyboots216) [reply core/fed_avg_optimizer.py]: Yep -this occurs only when the initialization_payload ends up being a Transaction object.

File: core/utils/dmljob.py (hunk 2 above)
Comment (kiddyboots216) [reply core/utils/dmljob.py]: Nice catch gotcha

File: core/utils/dmljob.py (hunk 3 above)
Comment (kiddyboots216) [reply core/utils/dmljob.py]: Nope! Decided against it

File: tests/testing_utils.py (hunk 4 above)
Comment (kiddyboots216) [reply tests/testing_utils.py]: blz

File: core/fed_avg_optimizer.py (hunk 1 above)
Comment (kiddyboots216) [reply core/fed_avg_optimizer.py]: Just going to delete this actually.

**Pull request merged**